web-admin/src/features/bookmarks/selectors.ts

```diff
@@ -259,8 +248,13 @@ function isFilterOnlyBookmark(
   metricsViewSpec: V1MetricsViewSpec,
   exploreSpec: V1ExploreSpec,
   timeRangeSummary: V1TimeRangeSummary | undefined,
-  defaultExploreUrlParams: URLSearchParams,
 ): boolean {
+  // This is needed as fix for bookmarks that set selectedComparisonDimension=undefined instead of selectedComparisonDimension=""
+  // We cant update fromProto since it will break backwards compatibility
+  if (!bookmarkState.selectedComparisonDimension) {
+    delete bookmarkState.selectedComparisonDimension;
+  }
```

Comment:
I'm not following – why can't we handle this case in `fromProto`?

Reply:
We have this in toProto,
```
 if (metrics.selectedComparisonDimension) {
   state.comparisonDimension = metrics.selectedComparisonDimension;
 }
```
So when selectedComparisonDimension is unset it would be `""` and `state.comparisonDimension` would be undefined. But if `metrics.selectedComparisonDimension` is undefined, in the case of filter-only bookmark we still set it as undefined. If we treat undefined in fromProto as not set then there would be backwards compatibility issues with old complete bookmarks, they would fail to replace current `selectedComparisonDimension`